pages/sites/[slug]/[locale]/index.tsx

```diff
@@ -91,10 +88,9 @@
 
   // Deprecation Notice: This route will be removed in next major version
   React.useEffect(() => {
-    if (router.query.p) {
-      router.push('/[p]', `/${router.query.p}`, {
-        shallow: true,
-      });
+    if (typeof router.query.p === 'string') {
+      const safePath = encodeURIComponent(router.query.p);
+      router.push(encodeURI(`/${safePath}`));
```

Comment:
The route `/[p]` will check that the project exists, otherwise the user will be shown a Resource not found error and directed to the index route. Additional validation isn't needed.